### DISCUSSION TITLE
[CONTENT] Introduction of input in non English languages in L18

### DISCUSSION THREAD

**Discussion by clementcontet** [hedyorg/hedy#4727] [Language Ideas]
**Describe the issue**
At the end of L18, one reads: **The final change we will need to make to get Python code is changing `ask` into `input`.**
When translated, some languages added a new word for `input` (like `invoer` in Dutch), other used the same word as for `ask` (like `demande` in French).
In the first case, it may confuse the student: is the new word regular Python code? If not, why introduce it?
In the second case, it doesn't make sense at all (**changing `demande` into `demande`**).

**Expected text**
I don't know if it is possible, but in all languages but English, could we have, from the beginning, `input` instead of `ask` as the English command counterpart, and skip completely the last sentence about the final change ?

**Comment by Felienne** [comment]
> **Describe the issue** At the end of L18, one reads: **The final change we will need to make to get Python code is changing `ask` into `input`.** When translated, some languages added a new word for `input` (like `invoer` in Dutch), other used the same word as for `ask` (like `demande` in French). In the first case, it may confuse the student: is the new word regular Python code? If not, why introduce it? In the second case, it doesn't make sense at all (**changing `demande` into `demande`**).

Ah this is an interesting point @clementcontet! We did not always have translated keywords, so this made sense in that situation. I agree it is a bit weird now to change it into something that is also not Python! We could not translate input and that would solve the issue, but that is a bit of a technical hassle and there are other reasons that that would not be preferred (adding one English word for non-English users) so I think for now we will leave this as is, but it is an interesting point, so I will convert it to a discussion!


> **Expected text** I don't know if it is possible, but in all languages but English, could we have, from the beginning, `input` instead of `ask` as the English command counterpart, and skip completely the last sentence about the final change ?

I know this is not the main point, but this is a deliberate choice, `input` in a confusing word (can be confused with iMput, and what does it even mean? So I do like ask for the beginning.

**Reply by clementcontet** [reply]
> this is a deliberate choice

I totally understand why introducing `ask` before `input` makes the learning curve easier in English!

My idea was to keep both for English users, but skip `ask` for others: have only one keyword (that would be the translation of ask) that would be translated as `input` in the commands popup, and skip the last sentence about final change for these users.
But I agree that it would be technically complicated for a small benefit!
